### DISCUSSION TITLE
How do I know if any memory leaks in my codes?

### DISCUSSION THREAD

**Discussion by LvJC** [plasma-umass/scalene#283] [Q&A] [answered]
Don't know how to use scalene to check if any memory leaks happen in my codes.

**Comment by emeryberger** [comment, marked as answer]
The top line indicates whether memory is growing over time. The memory trend lines for each line should show growth as an uphill slope, unless it's very slow. Scalene now also features an experimental flag that may help: `--memory-leak-detector` -- this is intended to identify lines that are responsible for allocating memory that leaks.
